Data/Storable/Endian.hs

```diff
@@ -11,31 +13,40 @@
 -- Storable instances with endianness.
 --
 
-module Data.Storable.Endian 
+module Data.Storable.Endian
   (
     LittleEndian(..), BigEndian(..),
     HasLittleEndian(..), HasBigEndian(..)
-  ) 
+  )
   where
 
 import System.ByteOrder
 
-import Foreign.Storable
 import Foreign.Ptr
+import Foreign.Storable
 
 import Unsafe.Coerce
 
-import Data.Word
 import Data.Bits
+import Data.Word
 
 #if defined(__GLASGOW_HASKELL__) && !defined(__HADDOCK__)
 import GHC.Base
-import GHC.Word
 import GHC.Int
+import GHC.Word
 #endif
 
+import GHC.Generics (Generic(..))
+
 newtype LittleEndian a = LE a
+  deriving ( Eq, Ord, Show, Read, Enum
+           , Num, Integral, Fractional, Floating, Real, RealFrac, RealFloat
```

Comment:
I'm concerned that this looks like "derive anything in the standard library that can possibly be derived", which (technically) should be maintained as more things keep being automatically derivable.

Is there some reasonable principle by which you can limit this set? I'm not quite what is the context in which using, say, only `instance RealFrac a` leads to boilerplate code, but having `instance RealFrac (LittleEndian a)` saves the day.